### DISCUSSION TITLE
Flaky tests on Xcode 14.3 while working fine on Xcode 14.2 (related to Swift 5.8)??

### DISCUSSION THREAD

**Discussion by 3a4oT** [pointfreeco/swift-composable-architecture#2028] [Q&A] [answered]
Hello, I have tried to run tests in Xcode 14.3 (CI is still on Xcode 14.2), and some tests fail occasionally. On Xcode 14.2, everything is fine. `TCA version 0.52.0`. Have anyone faced with same issues? I have mixed old-style reducers and new reducers wrapped with `AnyReducer` if that matter. Most failures related to `Received unexpected action`.

**Comment by mbrandonw** [comment]
Hi @3a4oT, tests that use time-based asynchrony with clocks/schedulers definitely seem to have gotten flakier in Swift 5.8. [Reliably writing tests](https://forums.swift.org/t/reliably-testing-code-that-adopts-swift-concurrency/57304) for Swift's concurrency tools continues to be incredibly difficult.

We do have some ideas of how to improve the situation, but for now you may need to sprinkle in some `await Task.yield()`s or very short `Task.sleep`s to push forward.

**Reply by 3a4oT** [reply]
Thanks for the fast reply. Unfortunately, neither `await Task.yield()` nor `Task.sleep` helped me. However, I used [SerialExecutor](https://forums.swift.org/t/reliably-testing-code-that-adopts-swift-concurrency/57304/81) and wrapped flaky tests with it, and so far, no more false positive tests.  While this is an unfortunate workaround, it seems to be acceptable to me in the short term.

PS. Thank you for your hard work and for saving tons of time for other developers, every time I am faced with an issue, there is an answer.

**Reply by mbrandonw** [reply, marked as answer]
Oh great, thanks for giving the serial executor a shot, and great to here it is actually working for you! That is exactly what we plan on baking directly into `TestStore` eventually, so hopefully these tricks won't be necessary for long.
